### PR TITLE
feat: listing card takes optional children

### PR DESCRIPTION
**Pull request by emilyjablonski**
# Pull Request Template

## Issue Overview

This PR addresses #2406

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

ListingCard now takes optional children to support custom content where the tables currently live - before, it had to display a table. The code changes are minimal, the story is to just POC the ticket with our built in styles. That section would only live with a consumer.

Ideally in the future I think both of these pieces should not be tied together and we can simplify the ListingCard interface but this is more time-sensitive.

## How Can This Be Tested/Reviewed?

Look at the [new stories](https://deploy-preview-2407--dev-storybook-bloom.netlify.app/?path=/story/listing-listingcard--custom-content) in ListingCard

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: ui-components/src/page_components/listing/ListingCard.stories.tsx [hunk 1]
```diff
@@ -77,3 +79,76 @@ export const withStackedTable = () => {
...
+      }}
+    />
+  )
+}
+
+const exampleCustomContent = () => {
+  const getHeader = (header: string) => {
+    return (
+      <div
+        className={
```
Comment (emilyjablonski) [inline ui-components/src/page_components/listing/ListingCard.stories.tsx:101]: Would love to pull our combinations of text classes like these more parent classes so we can standardize more variations of "text-header-1" etc

**Review by akegan — APPROVED** [review]
LGTM other than the comment in slack about adding more space. Thanks!

File: ui-components/src/page_components/listing/ListingCard.tsx [hunk 2]
```diff
@@ -18,41 +18,51 @@ export interface ListingCardHeaderProps {
...
-          >
-            {tableHeaderProps?.tableSubHeader}
-          </h4>
-        )}
+        {tableHeaderProps?.tableHeader && tableHeader()}
+        {tableHeaderProps?.tableSubHeader && tableSubHeader()}
         <div className="listings-row_table">
-          {(tableProps.data || tableProps.stackedData) && (
+          {children && children}
+          {tableProps && (tableProps.data || tableProps.stackedData) && (
```
Comment (jaredcwhite) [inline ui-components/src/page_components/listing/ListingCard.tsx:65]: Just a stylistic thing…maybe do `{(tableProps?.data || tableProps?.stackedData) && ( …` instead?

**Review by jaredcwhite — APPROVED** [review]

**Comment by jaredcwhite** [comment]
@akegan Hey sorry about the delayed response, somehow I thought I'd already replied but apparently not! Anyway, this should be good to go…the failed cypress-partners test is another issue from the dev branch that I think either has or will soon be resolved separately.

**Comment by akegan** [comment]
ok thanks @jaredcwhite. I don't have permission to merge with failing CI tests, so please merge when you can so I can pull it in, thanks

**Pull request merged**